app/models/concerns/spree/gift_cards/order_concerns.rb

```diff
@@ -11,17 +11,22 @@ module GiftCards::OrderConcerns
     end
 
     module InstanceMethods
-      def finalize!
-        create_gift_cards
-        super
+      def gift_card_match(line_item, options)
+        gift_card_options = options["gift_card_details"]
+        !line_item.gift_card? ||
+          (line_item.gift_card? && gift_card_options.present? &&
```

Comment:
would this be able to be simplified as:

```ruby
  return true unless line_item.gift_card?
  line_item.gift_cards.any? { |gift_card| gift_card.details == options["gift_card_details"] }
```

if you push https://github.com/solidusio/solidus_virtual_gift_card/pull/9/files#diff-94036a0cdf611599347aed2b6e700caaR18 down to the gift card model?